apps/main_app/lib/main_app.ex

```diff
@@ -3,19 +3,25 @@ defmodule MainApp do
 
   use Application
 
-  @subscriber Application.compile_env(:main_app, :subscriber, nil)
   @replicator Application.compile_env(:main_app, :replicator, nil)
-  @publisher Application.compile_env(:main_app, :publisher, nil)
+  @publishers Application.compile_env(:main_app, :publishers, [])
 
   @impl true
   def start(_type, _args) do
     children = [
-      @subscriber,
+      MainApp.Communicator,
+      Enum.map(@publishers, fn %{name: name, init_arg: init_arg, module: module} ->
+        %{
+          id: name,
+          start: {module, :start_link, [name, init_arg]}
+        }
+      end),
       {@replicator,
```

Comment:
Maybe should specify `type: :supervisor`? https://hexdocs.pm/elixir/Supervisor.html#module-child-specification

Reply:
Actually I think that this one is set automatically by the fact that `@replicator` has `use Supervisor`. As you can see in the doc: https://hexdocs.pm/elixir/Supervisor.html#module-module-based-supervisors (at the bottom of this section) we can customize only `id` and `restart` for the generated child_spec/1 function (it would be kind of unintuitive to set `:worker` for Supervisor child spec).